feature/integrate gcs kms bucket for storing images

ercsnmrs opened the pull request:
Overview:
- KMS-protected buckets gives full control over encryption keys (vs Google-managed keys), granular permissions: You can restrict access to specific keys per team or service, and rotation & auditability: You can rotate keys periodically and audit access using Cloud Logging.

ercsnmrs committed:
feature/integrate gcs kms bucket for storing images

Overview:
- KMS-protected buckets gives full control over encryption keys (vs Google-managed keys), granular permissions: You can restrict access to specific keys per team or service, and rotation & auditability: You can rotate keys periodically and audit access using Cloud Logging.

ercsnmrs commented:
implemented the same approach on an enterprise solution 
- I will test it out first by removing the limitation on maxInbandAttachmentSize on the Webapp
- I will post the results here.

or-else commented:
Cool, thanks.

> removing the limitation on maxInbandAttachmentSize on the Webapp

I'm not sure what you mean by this. Do you want to set `maxInbandAttachmentSize` to zero?

ercsnmrs commented:
@or-else i am trying to test the upper limit of the integration (e.g rps on which the integration to kms buckets break).

One use case i see here is that we can offload all images and attachment to the cloud instead of saving to db. I see it as a path for optimization here. 

- [ ] To do this, we need to send everything to gcs by lowering the maxInBandAttachmensize or disabling (i will do this locally)

or-else commented:
OK, makes sense.

or-else commented:
Should I review this PR too?

or-else committed:
Update README.md

Mention article where Tinode is claimed to be the backend for the Russian National Messenger Max

or-else committed:
Update README.md

Typo

or-else committed:
Update README.md

Remove text added by mistake
